internal/nvidia/nvidia.go

```diff
@@ -0,0 +1,130 @@
+package nvidia
+
+import (
+	"fmt"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	"github.com/markusressel/fan2go/internal/configuration"
+	"github.com/markusressel/fan2go/internal/fans"
+	"github.com/markusressel/fan2go/internal/sensors"
+)
+
+type NvidiaController struct { // TODO: why "controller"?
+	Identifier string // e.g. "nvidia-10de2489-0400"
+	Name       string // e.g. "NVIDIA GeForce RTX 3060 Ti"
+
+	Fans []fans.NvidiaFan
+	// at least currently nvml only supports one temperature sensor
+	// (pointer in case no sensor was found at all)
+	Sensor *sensors.NvidiaSensor
+}
+
+// create identifier for nvidia devices (for config), which is like
+//
+//	nvidia-<PCI vendor ID><PCI device ID>-<PCI address>
+//
+// where PCI address is calculated in the same way as libsensor PCI device addresses
+// example: "nvidia-10de2489-0400"
+func getDeviceID(device nvml.Device) string {
+	pciInfo, ret := device.GetPciInfo()
+	if ret != nvml.SUCCESS {
+		// shouldn't really happen - display error?
+		return ""
+	}
+	var pciVendorID uint16 = uint16(pciInfo.PciDeviceId & 0xFFFF)
+	var pciDeviceID uint16 = uint16((pciInfo.PciDeviceId >> 16) & 0xFFFF)
+	// TODO: if the PCI "function" value is really needed, it could be parsed from pciInfo.BusId, I guess
+	//  for now I assume that 0 always works? (on my card function 1 is the soundcard for HDMI audio => not relevant here)
```

Comment:
The DeviceID is only used to match the config entry to the actual hardware device. As long as it is unique to the correct hardware device, we don't necessarily need to include every field available, only the ones that are required to make it unique.

Reply:
Currently my assumption is that this is not needed, but I only have a single card to test.
But regarding my previous point about future-proofing, I should probably just remove this ;)
I mostly added it for symmetry with lmsensors IDs

Reply:
Part of the IDs used for the hwmon stuff also used to not include everything, until it became necessary, and then I aligned it with the output of lmsensors :sweat_smile: 

I have no preference here, use what works :man_shrugging: